### PR TITLE
fix(server_family): Remove search indexes during the FLUSHALL command

### LINKED ISSUE #3532: FLUSHALL command does not remove indexes
`FLUSHALL` command does not remove indexes. 

**Reproducible Code Snippet**
```bash
127.0.0.1:6379> FT.CREATE json ON JSON SCHEMA $.nested.value AS value TEXT
OK
127.0.0.1:6379> flushall
OK
127.0.0.1:6379> FT.CREATE json ON JSON SCHEMA $.nested.anothervalue AS value TEXT
(error) ERR Index already exists
```

### PR DESCRIPTION
**Pull request by BagritsevichStepan**
fixes #3532 and #3592

### CODE REVIEW

File: src/server/server_family.cc [hunk 1]
```diff
@@ -1577,6 +1577,9 @@ error_code ServerFamily::Drakarys(Transaction* transaction, DbIndex db_ind) {
   transaction->Execute(
       [db_ind](Transaction* t, EngineShard* shard) {
         t->GetDbSlice(shard->shard_id()).FlushDb(db_ind);
+        if (!db_ind || db_ind == DbSlice::kDbAll) {
+          shard->search_indices()->DropAllIndices();
+        }
         return OpStatus::OK;
```
Comment (dranikpg) [inline src/server/server_family.cc:1583]: 1. Can't we call this from FlushDb?
2. Let's try getting rid of this part then https://github.com/dragonflydb/dragonfly/blob/54cf6f0d06dff9b7ffb2dad2c92a2a69e3a0363e/src/server/rdb_load.cc#L2625-L2634

File: src/server/rdb_load.cc [hunk 2]
```diff
@@ -2627,10 +2627,8 @@ void RdbLoader::PerformPreLoad(Service* service) {
   if (cmd == nullptr)
     return;  // MacOS
 
-  shard_set->AwaitRunningOnShardQueue([](EngineShard* es) {
-    for (const auto& name : es->search_indices()->GetIndexNames())
-      es->search_indices()->DropIndex(name);
-  });
+  shard_set->AwaitRunningOnShardQueue(
+      [](EngineShard* es) { es->search_indices()->DropAllIndices(); });
```
Comment (dranikpg) [inline src/server/rdb_load.cc:2631]: And now please validate than FlushAll which is called before replication and loading a dataset cleans the indices. The initial goal of moving the drop into FlushDb was to get rid of this code entirely 🙂

File: src/server/server_family_test.cc [hunk 3]
```diff
@@ -545,4 +545,33 @@ TEST_F(ServerFamilyTest, ConfigNormalization) {
               RespArray(ElementsAre("replica_priority", "13")));
 }
 
+TEST_F(ServerFamilyTest, FlushSearchIndices) {
```
Comment (romange) [inline src/server/server_family_test.cc:548]: I think it belongs to search_family_test even though the bug is related to flush

Comment (BagritsevichStepan) [reply src/server/server_family_test.cc:548]: Yes, I also thought to which test it would be better to add it

**Review by romange — APPROVED** [review]

**Pull request merged**